Add Galaxy as Tool assembly

beatrizserrano opened the pull request:
Hi all,

As discussed during the BH2021 (long time ago :sweat_smile:) this PR is to add the first basic page for Galaxy as Tool assembly. We have created an image with all the solutions that Galaxy provides across the data life cycle using the default from the RDMkit adding the Galaxy logo. We will also present this in a poster at ECCB2022 in September. Can we use the image of the life cycle in general?

Thank you!

cc @bgruening

bedroesb commented:
@beatrizserrano I love the design of the image! Looks really pro. Little feedback: I don't know if I can figure out only based on the image one one step is with a dotted line, and the Plan step in white (inverted). 

PS Yes you can reuse it, we do like though if you add somewhere our little logo https://rdmkit.elixir-europe.org/media_kit (it can also be just the R :)

beatrizserrano commented:
Thanks @bedroesb! I've added the logos to the center of the image. 

> I don't know if I can figure out only based on the image why one step is with a dotted line, and the Plan step in white (inverted).

We've removed the dotted line. The _Plan_ step is inverted because Galaxy doesn't offer any feature for that. If you think it's still confusing, I can leave the original orange :)

bedroesb commented:
Well I like the idea of making it different from the others, but I don't know if inverting it is the nicest, I was playing around with some other ways: 
![galaxy-rdmkite](https://user-images.githubusercontent.com/44875756/182167284-0d983df7-c47d-42ba-a679-9413974cf333.png)
![galaxy-rdmkit2](https://user-images.githubusercontent.com/44875756/182167602-a7b0bbb2-fccb-4555-ba84-19c2e12ed3c6.png)

But I don't know if I like them more, @martin-nc, @korbinib  if you have time, could you give your opinion on this :)

beatrizserrano commented:
grey looks nice to me :)

martin-nc commented:
Since Bert has called me in, I'll give a comment, but feel free to ignore!

1. What's the purpose of the illustration? If it's to give a brief overview of how Galaxy can help you through the data life cycle, then I think there's too much text. If it's to give more details, then maybe an illustration isn't the best way to convey this information?
2. How important is the text? If it's important then I guess it should be in HTML rather than an image, so it can be read by search engines and assistive devices, and it is easier to read on small screens. 

If the answer to 1 is "A brief summary", I'd do something simpler like this:

<img src="https://user-images.githubusercontent.com/15261745/183649080-7841a610-03aa-4ba4-9d7b-3a7d7030fc66.png" width=75% alt="RDMlife cycle" />

I've only done three sections, but hopefully you'll get the gist.

Or you could think mobile-first and do a title like "How Galaxy helps you through the stages of the RDM life cycle", then have tiles instead. These would form columns on wider screens:

<img src="https://user-images.githubusercontent.com/15261745/183647789-b83312dd-1c93-4e02-81c8-d5c7807d38c4.png" width=35% alt="RDMlife tiles" />

If the details are important, then the tiles can be extended, and again form columns on wider screens:

<img src="https://user-images.githubusercontent.com/15261745/183650277-7bb9b27b-cdbb-4b29-a5dd-5c8244d9fe62.png" width=35% alt="Extended tiles" />

Anyway those are just some ideas.

beatrizserrano commented:
Thank you @martin-nc, the accessibility is a very valuable point. I like the last version that you presented, the tile including the details. Do you have an example for me to get some inspiration on how to implement this idea?

martin-nc commented:
@beatrizserrano maybe I was getting too creative there, because it's not something we can do simply in markdown! If you're okay fiddling with Bootstrap classes, though, then something like this:

```
<div class="container-fluid g-0 my-2">
    <div class="row row-cols-1 row-cols-md-3">
        <div class="col">
            <div class="card border border-2">
                <div class="card-body px-4">
                    <h3 class="card-title text-center mb-4 mt-3">Collect</h3>
                    <h4>Access to databases</h4>
                    <ul class="lh-sm">
                        <li>UNIPROT</li>
                        <li>INTERMINE</li>
                        <li>...</li>
                    </ul>
                    <h4>Customised data access</h4>
                    <ul class="lh-sm">
                        <li>Text 1</li>
                        <li>Text 2</li>
                    </ul>
                    <h4>LIMS integration</h4>
                    <ul class="lh-sm">
                        <li>Text 1</li>
                        <li>Text 2</li>
                    </ul>
                </div>
            </div>
        </div>
        <div class="col">
            <div class="card border border-2">
                <div class="card-body px-4">
                    <h3 class="card-title text-center mb-4 mt-3">Process</h3>
                    <h4>Data transformation</h4>
                    <ul class="lh-sm">
                        <li>UNIPROT</li>
                        <li>INTERMINE</li>
                        <li>...</li>
                    </ul>
                    <h4>Import workflows</h4>
                    <ul class="lh-sm">
                        <li>Text 1</li>
                        <li>Text 2</li>
                    </ul>
                    <h4>Metadata handling</h4>
                    <ul class="lh-sm">
                        <li>Text 1</li>
                        <li>Text 2</li>
                    </ul>
                </div>
            </div>
        </div>
        <div class="col">
            <div class="card border border-2">
                <div class="card-body px-4">
                    <h3 class="card-title text-center mb-4 mt-3">Analyse</h3>
                    <ul>
                        <li>A bit of blurb.</li>
                    </ul>
                </div>
            </div>
        </div>
    </div>
</div>
```

Should produce something like this:
![Screenshot 2022-08-10 at 17 53 05](https://user-images.githubusercontent.com/15261745/183973568-e2693418-49f2-4b55-bb0d-3973200cee66.png)

@bedroesb can confirm, but I don't think we have an easy way to add the life cycle colours. We'd have to create some custom classes, or maybe it's enough to add the icon of the right colour with the header. We have the icons around somewhere.

If this seems a faff, though, then I'd just go with the original diagram. Maybe just have a caption with more explanatory text for SEO/accessibility.

beatrizserrano commented:
I'll give it a try and we compare which version looks nicer :) thanks @martin-nc

beatrizserrano commented:
This is the best that I could do, used background colours with some transparency:

![image](https://user-images.githubusercontent.com/11300231/184651550-f6876a04-c2c2-4937-ac0a-e5846dd18500.png)

Not crucial details, but I see a couple of improvements:
- I didn't manage to have rounded corners, any suggestions on how to do that?
- It would be nice to add the icons. I see that they are [here](https://thenounproject.com/ELIXIRCommunications/kit/rdmkit/#). How do I add them?

martin-nc commented:
For the rounded corners, I think you can just use the `rounded` class on the card: `<div class="card mb-4 rounded">`.

 I'm not sure about the tinted backgrounds: is there enough contrast ratio between the text and background? https://webaim.org/resources/contrastchecker/. That's why I suggested the border and outline being in colour and the background white. The white background is usually a bit safer for contrast and for colour blind users, but maybe it's okay.

As for the icons, here are some you can put in the images directory to experiment with: [life-cycle-icons.zip](https://github.com/elixir-europe/rdmkit/files/9338649/life-cycle-icons.zip).

There are many ways to include the images. Maybe it's simplest to include them like a normal image:
```
<div class="card-body px-4" style="background-color: #fac54b45">
 <div class="w-25 mx-auto"><img src="/images/life-cycle-icons/collect-icon.svg" alt="" /></div>
          <h3 class="card-title text-center mb-4 mt-3">Collect</h3>
```
The icon is purely decorative so I've left the `alt` attribute empty.

Maybe @bedroesb has thoughts on this, though, since he's more familiar with the site code. Also, @korbinib for the format of the tool assembly pages.

bedroesb commented:
@beatrizserrano and @martin-nc I took the freedom to update a little bit the html code :) Following the suggested syntax from [bootstrap](https://getbootstrap.com/docs/5.2/components/card/#grid-cards) for grid cards, you can get rid of many padding and margin classes + it is rounded by default, in the the card class. I also updated you branch with the remote because it was a little bit lagging behind ;) We for example took away the audience attribute etc. I will try to add the icons and have a look!
![Screenshot from 2022-08-22 11-16-27](https://user-images.githubusercontent.com/44875756/185885955-62632e77-d232-4aca-b6fe-50fa95c002ff.png)

bedroesb commented:
@beatrizserrano and @martin-nc, this is with the icons in black:
![Screenshot from 2022-08-22 13-28-29](https://user-images.githubusercontent.com/44875756/185910845-99cae1b1-ebff-46ce-bdda-082736188ed0.png)

beatrizserrano commented:
Nice! I like the icons. Maybe the card header (where the icon and the title are) could have the background colour. Or would it make it less accessible?

bedroesb commented:
If I understood correctly:
![Screenshot from 2022-08-22 16-31-06](https://user-images.githubusercontent.com/44875756/185946877-7cfb7f60-56c7-432b-8136-e8f06f6ae826.png)
![Screenshot from 2022-08-22 16-24-43](https://user-images.githubusercontent.com/44875756/185946881-3143aebb-24da-421a-8afa-a507b98c0f6b.png)

beatrizserrano commented:
Looks great to me. The first version better? What do people think?

bgruening commented:
I like the version with the stronger colors and white icons a bit more :)

bedroesb commented:
ok done! 
![Screenshot from 2022-08-23 10-14-10](https://user-images.githubusercontent.com/44875756/186107743-d19044b1-e3e7-45f2-b50f-7644c2d69bd7.png)

@elixir-europe/rdmkit-editors  what are your opinions?

bgruening commented:
Cool, very nice! Thanks, everyone for working on this!

korbinib commented on _data/affiliations.yaml:
```diff
@@ -136,3 +136,9 @@
   type: infrastructure
   url: https://nels.bioinfo.no/
+- name: European Union
+  image_url: /images/funders/Flag_of_Europe.svg
```
shouldn't it be the EC logo instead of the EU flag, as the doi is pointing to the EC?

E.g. https://ec.europa.eu/info/publications/standard-european-commission-logos_en

bedroesb replied:
You are right, I updated it :)

korbinib commented on pages/tool_assembly/galaxy_assembly.md:
```diff
@@ -0,0 +1,220 @@
+        <h4>Metadata handling</h4>
+        <ul class="lh-sm">
+          <li>Provenance tracking</li>
+          <li>Automatic metadata enrichment</li>
```
I feel this could be more extensive

bedroesb replied:
@beatrizserrano any comments?

beatrizserrano replied:
Not much from my side, I think. I don't know if there's something generic enough to be specified here, what do you think @bgruening?

bedroesb replied:
@korbinib if @bgruening gives green light, and you give your approval, you can merge. I am on holiday tomorrow :)

bgruening replied:
We could add here more (like datatypes, EDAM annotaitons, number of columns, reference genome ...), the question is just if it's worth it and how. I guess for the first PR this is good and we keep adding things to those list later. :+1: from my side. Lets get it in :)

korbinib reviewed (commented):
Could you check about the EC logo vs EU flag?

bedroesb commented:
@korbinib the EC logo was already in place, so I think at one point I wanted to use that one :)

beatrizserrano commented:
Thank you all for the help!
